### ISSUE TITLE
feature: --wordlist-reverse

### ISSUE THREAD

**Issue by roycewilliams** [hashcat/hashcat#3116]
**Describe the feature**
`--wordlist-reverse` would read a wordlist from the _end_ of the file, instead of the beginning.

This would be especially useful in cases where a wordlist file is regularly appended to.

It could also be useful for crude meet-in-the-middle strategies.

**Additional context**
This new option would only work for files, not input from stdin - so a fatal error would need to be generated if stdin was attempted in combination with this option.

**Comment by b0lek** [comment]
I did a quick check and it looks quite hard to implement as it requires first reading whole file and then provide words in reverse order. Current code structure is prepared to read file from beginning till the end.

Why don't you reverse a file before running `hashcat`?

On Linux there is a simple command to do this `tac wordlist.txt > wordlist_reversed.txt`. 
`tac` is doing a job.

**Comment by roycewilliams** [comment]
It most definitely does not require reading the entire file first. It can function like `tac` does under the hood - reading from the end of the file in blocks.
If this is natively supported by hashcat, I neither have to double my disk usage for wordlist files, nor dramatically slow down hashcat performance for fast hashes by forcing the wordlist to be piped in from `stdin`.

**Comment by b0lek** [comment]
Yes it can be done without reading whole file but anyway I don't see an easy way to implement this.
If I will find a way to safely implement this than maybe I'll get back to this idea. 

How long wordlists are you using?
